### PR TITLE
Configure sonarcloud and travis-ci

### PR DESCRIPTION
**Pull request by rsynek**
- configures jacoco code coverage and reporting to sonarcloud.io
- enables travis-ci to update the sonarcloud.io with every PR

### CODE REVIEW

**Comment by rsynek** [comment]
@ge0ffrey please review

File: pom.xml [hunk 1]
```diff
@@ -90,6 +96,11 @@
         <!-- Old version until this issue is resolved: https://github.com/ImmobilienScout24/illegal-transitive-dependency-check/issues/30 -->
         <version>1.0.4</version>
       </dependency>
+      <dependency>
+        <groupId>org.jacoco</groupId>
+        <artifactId>org.jacoco.agent</artifactId>
+        <version>${version.jacoco.plugin}</version>
```
Comment (ge0ffrey) [inline pom.xml]: I'd expect that jacoco is a compile/build/tool dependency, not a real normal dependency.
Is there any way we can restrict jacoco to one (or more) plugin classpaths?

File: pom.xml [hunk 2]
```diff
@@ -106,6 +117,84 @@
...
+              <artifactId>gwt-maven-plugin</artifactId>
+              <configuration>
+                <skipCompilation>true</skipCompilation>
+              </configuration>
+            </plugin>
+            <plugin>
+              <artifactId>maven-surefire-plugin</artifactId>
+              <configuration>
+                <argLine>${surefire.argLine}</argLine>
+              </configuration>
```
Comment (ge0ffrey) [inline pom.xml]: What if we add a dependencies element here with the jacoco dependency?

File: pom.xml [hunk 3]
```diff
@@ -28,6 +28,12 @@
 
   <properties>
     <findbugs.failOnViolation>true</findbugs.failOnViolation>
+    <jacoco.agent.line/>
+    <surefire.argLine>
+      -Dfile.encoding=${project.build.sourceEncoding}
+      ${jacoco.agent.line}
+    </surefire.argLine>
+    <version.jacoco.plugin>0.8.2</version.jacoco.plugin>
```
Comment (ge0ffrey) [inline pom.xml]: Can we only overrides these properties in the jacoco profile?

Comment (rsynek) [reply pom.xml]: +1, even as just a temporary solution - it will end up in kie-parent

File: pom.xml (hunk 1 above)
Comment (rsynek) [reply pom.xml]: It makes sense, will look into it.

**Comment by rsynek** [comment]
@ge0ffrey changed the jacoco to the surefire plugin's dependency as you suggested. Thanks for this idea. Ready to merge, once the jenkins build passes too.

**Pull request merged**